### ISSUE TITLE
[Suggestion] Consider making the DO_NOT_ORG_ROAM tag into a property

### ISSUE THREAD

**Issue by ifinkelstein** [dmitrym0/org-hyperscheduler#8]
**Issue:**
The `DO_NOT_ORG_ROAM` tag is visually distracting as it appears in most `org-agenda` views. Hiding tags can be impractical as many of us use tags in our daily workflows.

**Proposal:**
Consider adding `DO_NOT_ORG_ROAM` as a custom property instead of a tag. This property can remain hidden in the properties drawer but can still be accessed by `org-roam-db-node-include-function`
[org property drawer API](https://orgmode.org/manual/Using-the-Property-API.html)

**Comment by maikol-solis** [comment]
This is a great idea. 

I have the `org-roam` folder and the `gtd` one completely separated. The tag creates a little noise into my already defined tags (also, it's the biggest one). 

Maybe the tag/property could be activated only if the agenda file is inside the `org-roam-directory`. WDYT?

![image](https://user-images.githubusercontent.com/4078489/167254500-a785e29f-199e-4aff-8859-9c141a1cadf4.png)

**Comment by dmitrym0** [comment]
I had the same problem, this is how I dealt with it:

```emacs-lisp
(setq org-agenda-hide-tags-regexp (regexp-opt '("DO_NOT_ORG_ROAM" "MINE" "WORK" "IMPORTANT")))
```

> Regular expression used to filter away specific tags in agenda views.
> This means that these tags will be present, but not be shown in the agenda
> line.  Secondary filtering will still work on the hidden tags.
> Nil means don’t hide any tags.

Your approach is better since it doesn't require any additional work. I'll add it to the "roadmap".

**Comment by ifinkelstein** [comment]
> I have the `org-roam` folder and the `gtd` one completely separated. The tag creates a little noise into my already defined tags (also, it's the biggest one).
Same.

> Maybe the tag/property could be activated only if the agenda file is inside the `org-roam-directory`. WDYT?
An optimal solution may be to have a user-accessible setting, maybe `(setq org-hs-add-org-roam-ignore t)'

I also think that this setting is more in-line with a property drawer than a tag. 

Thanks for the ProTip on how to hide unwanted tags!

**Comment by dmitrym0** [comment]
Hey @meliache, thanks for your suggestion. I spent a few minutes working on this today before actually seeing your comment.

In [ea2aad3](https://github.com/dmitrym0/org-hyperscheduler/commit/ea2aad3461a6f5d0509e593a4418bd8295d24a88), I removed the `DO_NOT_ORG_ROAM` tag in favour of drawer properties like @ifinkelstein proposed.

I like your hooks approach, but I'm wondering if it's too complicated. Do you have any other use cases in mind, where folks might want to use the hook? I'm leaning towards a flag (like [this](https://github.com/dmitrym0/org-hyperscheduler/issues/8#issuecomment-1120323810)). Toggling a flag is much less error prone than having a hook.

**Comment by meliache** [comment]
@dmitrym0 No, I don't really have any other use-cases for the hook in mind. Though often the idea for those is to enable users to add behaviours beyond what the developer could imagine. In theory you could have both a flag and a hook, e.g.
``` emacs lisp
(when org-hs-add-org-roam-ignore
    (add-hook #'org-hyperscheduler-post-id-hook (lambda () (org-set-tags (org-uniquify (cons "DO_NOT_ORG_ROAM" (org-get-tags)))))))
 ```
 But in this case maybe you're right to better keep it simple and just add a flag, here I also don't think I hook is really necessary. And if anyone wants to completely change the behaviour, it's emacs, they can just override or advice it.

**Comment by dmitrym0** [comment]
This is now fixed in v0.1.3+4. If you get the latest version, you shouldn't have to do anything; by default adding `ROAM_EXCLUDE` property is disabled. 

If you do need `ROAM_EXCLUDE` , `customize-group` or the usual `(set org-hyperscheduler-exclude-from-org-roam t)` should do the trick.
